frontend/apps/reader/modules/readerview.lua

```diff
@@ -482,6 +493,16 @@ function ReaderView:drawScrollView(bb, x, y)
         self.state.pos)
 end
 
+--- Converts a color name into a color struct
+function ReaderView:lookupHighlightColor(color_name)
+    if Screen:isColorEnabled() then
+        local color = self.highlight_colors[color_name]
+        return Blitbuffer.colorFromString(color or "#ffff00")
+    else
+        return Blitbuffer.COLOR_GRAY
```

Comment:
Maybe just return nil here, and do not check `Screen:isColorEnabled()` further to fallback.

Reply:
Ah I see your point now. `drawHighlightRect` already has `if not color` in all the blocks. So passing a nil color should be ok. Let me do that then.